tools/StaticAnalysis/BreakingChangeAnalyzer/TypeMetadataHelper.cs

```diff
@@ -370,6 +277,87 @@ public TypeMetadataHelper()
             TypeMetadata newTypeMetadata,
             ReportLogger<BreakingChangeIssue> issueLogger)
         {
+            int problemId = ProblemIds.BreakingChangeProblemId.ChangedOutputElementType;
+            string description = string.Format(Properties.Resources.ChangedOutputElementTypeDescription,
+                                    oldTypeMetadata.ElementType, newTypeMetadata.ElementType);
+            string remediation = string.Format(Properties.Resources.ChangedOutputElementTypeRemediation,
+                                    oldTypeMetadata.ElementType);
+
+            // If the type is an array, check for any breaking changes 
+            if (IsElementType(cmdlet, oldTypeMetadata, newTypeMetadata, problemId, description, remediation, issueLogger))
+            {
+                return;
+            }
+
+            // If the type is a generic, check for any breaking changes
+            if (oldTypeMetadata.GenericTypeArguments.Count > 0 && newTypeMetadata.GenericTypeArguments.Count > 0)
```

Comment:
It looks like this is repeated.  Is there any reason we can't refactor the equivalence check for GenericTypes into a single method?